app/javascript/src/utilities/json-utils.jsx

```diff
@@ -0,0 +1,21 @@
+// @flow
+
+// eslint-disable-next-line flowtype/no-weak-types
+const toJsonString = (val: Object): string => JSON.stringify(val, null, 2)
+
+const fromJsonString = (json: string) => JSON.parse(json)
+
+const safeFromJsonString = (json: string) => {
+  try {
+    return fromJsonString(json)
+  } catch (err) {
+    console.warn('That doesn\'t look like a valid json!', err)
```

Comment:
If we are expecting an error, why not throwing an actual error?
```suggestion
    throw new Error(err)
```